### ISSUE TITLE
GU0023 don't warn when used in lambda

### ISSUE THREAD

**Issue by JohanLarsson** [GuOrg/Gu.Analyzers#186]
```cs
namespace RoslynSandbox
{
    using System;

    public class Foo
    {
        public static readonly Func<int> Value1 = () => Value2;

        public static readonly int Value2 = 2;
    }
}
```
